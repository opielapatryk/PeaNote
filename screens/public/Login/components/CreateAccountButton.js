import React from 'react';
import {Pressable,Text} from 'react-native';
import { styles } from '../../../../assets/styles/styles';
import { signUpFirebase } from '../functions/signUpFirebase';
import { useSelector,useDispatch } from 'react-redux';

const CreateAccountButton = () => {
  const { email,password,first_name,last_name } = useSelector((state)=>state.login)
const dispatch = useDispatch()

    return (
<>
        <Pressable style={styles.confirmButton} onPress={()=>signUpFirebase(email,password,first_name,last_name,dispatch)}> 
          <Text>
            Get Started
          </Text>
        </Pressable>  
      </>
    );
}

export default CreateAccountButton;