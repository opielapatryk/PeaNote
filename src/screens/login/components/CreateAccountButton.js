import React from 'react';
import {Pressable,Text} from 'react-native';
import { styles } from '../../../../assets/styles/styles';
import { signUpFirebase } from '../functions/signUpFirebase';
import { useSelector,useDispatch } from 'react-redux';

const CreateAccountButton = () => {
  const { email,password } = useSelector((state)=>state.login)
  const dispatch = useDispatch()

  const onPress = () => {
    signUpFirebase(email, password, dispatch);
  };

  return (
    <>
      <Pressable style={styles.confirmButton} onPress={onPress}> 
        <Text style={{fontWeight:'600'}}>Get Started</Text>
      </Pressable>  
    </>
  );
}

export default CreateAccountButton;