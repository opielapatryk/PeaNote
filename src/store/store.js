import { configureStore } from '@reduxjs/toolkit'
import boardReducer from './notes/boardSlice'
import settingsReducer from './settings/settingsSlice'

export default configureStore({
    reducer:{
        board:boardReducer,
        settings:settingsReducer
    }
})