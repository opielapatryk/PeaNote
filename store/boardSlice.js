import { createSlice } from "@reduxjs/toolkit";

const initialState = {
    notes:[],
    isInput:true,   
    history:[]
}

export const boardSlice = createSlice({
    name:'board',
    initialState,
    reducers:{
        showInput: (state) =>{
            return {...state, isInput: true};
        },
        hideInput: (state) =>{
            return {...state, isInput: false};
        },
        addNote: (state, action) => {
            return {...state, notes:[...state.notes, action.payload],history: [...state.notes]}
        },
        removeNote: (state, action) => {
            return {
              ...state,
              notes: state.notes.filter(note => note.id !== action.payload),
              history: [...state.notes]
            };
          },
          changeInfo: (state, action) => {
            const prevText = state.history.find((note) => note.id === action.payload)?.text || "";
            return {
                ...state,
                notes: state.notes.map((note) => {
                    if (note.id === action.payload) {
                        return {
                            ...note,
                            isInfo: !note.isInfo,
                            text: note.isInfo ? prevText : 'click again to delete note',
                        };
                    }
                    return note;
                }),
                history: [...state.notes]
            };
        }     
    }
})

export const {showInput, addNote, removeNote, hideInput,changeInfo} = boardSlice.actions
export default boardSlice.reducer