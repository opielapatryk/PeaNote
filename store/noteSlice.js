import {createSlice } from '@reduxjs/toolkit'
const initialState = {
    id: 0,
    text:'',
    isNote:true,
    isInfo: false
}
export const noteSlice = createSlice({
    name:'note',
    initialState,
    reducers:{
        showInfo: (state)=>{
            return {...state,isInfo: true,text:'Click againt to delete note'}
        },
        setText: (state,action)=>{
            return {...state,text:action.payload}
        },
        hideInfo: (state)=>{
            return {...state,isInfo: false,text:initialState.text}
        },
        removeNote: (state)=>{
            return {...state,isNote: false}
        },
        isNoteTrue: (state)=>{
            return {...state,isNote: true}
        }
        
    }
})

export const {showInfo, hideInfo,removeNote,setText, isNoteTrue} = noteSlice.actions;

export default noteSlice.reducer