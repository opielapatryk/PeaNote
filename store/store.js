import { configureStore } from '@reduxjs/toolkit'
import boardReducer from './notes/boardSlice'

export default configureStore({
    reducer:{
        board:boardReducer
    }
})