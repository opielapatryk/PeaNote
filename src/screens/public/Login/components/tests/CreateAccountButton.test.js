import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';  
import store from '../../../../../store/store'; 
import CreateAccountButton from '../CreateAccountButton';

jest.mock('@react-native-firebase/auth', () => ({
    auth: {
        signInWithEmailAndPassword: jest.fn(),
        createUserWithEmailAndPassword: jest.fn(),
        signOut: jest.fn(),
    },
  }));
  
jest.mock('@react-native-firebase/firestore', () => ({
    firestore: {
        collection: jest.fn(),
        doc: jest.fn(),
        get: jest.fn(),
    },
}));

test('Create account button renders correctly.', () => {

  const tree = renderer.create(
    <Provider store={store}>
      <CreateAccountButton />
    </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
