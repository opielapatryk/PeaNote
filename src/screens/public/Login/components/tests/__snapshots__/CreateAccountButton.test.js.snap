// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create account button renders correctly. 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#c99c1f",
      "borderColor": "#c99c1f",
      "borderRadius": 5,
      "borderWidth": 1,
      "elevation": 1,
      "height": 66.7,
      "justifyContent": "center",
      "padding": 10,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.18,
      "shadowRadius": 1,
      "width": 500,
    }
  }
>
  <Text
    style={
      {
        "fontWeight": "600",
      }
    }
  >
    Get Started
  </Text>
</View>
`;
